src/sidebar/service-url.js

```diff
@@ -3,44 +3,91 @@
 var urlUtil = require('./util/url-util');
 
 /**
- * A map of all route names to relative URLs on the Hypothesis service that
- * the client links to.
+ * An AngularJS service that returns an absolute URL given a link name and
+ * params, by expanding named URL templates received from the annotation
+ * service's API.
  *
- * The URLs are relative to the `serviceUrl` URL in the app's settings.
+ * The links object from the API is a map of link names to URL templates:
+ *
+ * {
+ *   signup: "http://localhost:5000/signup",
+ *   user: "http://localhost:5000/u/:user",
+ *   ...
+ * }
+ *
+ * Given a link name (e.g. 'user') and params (e.g. {user: 'bob'}) return
+ * an absolute URL by expanding the named URL template from the API with the
+ * given params (e.g. "http://localhost:5000/u/bob").
+ *
+ * Before the links object has been received from the API this function
+ * always returns empty strings as the URLs. When the links object is
+ * received from the API it's updated in the Redux state store so that
+ * components that use this function will be re-processed, call this function
+ * again, and receive the real URL.
+ *
+ * @param {string} linkName - The name of the link to expand
+ * @param {object} params - The params with which to expand the link
+ * @returns {string} The expanded absolute URL, or an empty string if the
+ *                   links haven't been received from the API yet
+ * @throws {Error} If the links have been received from the API but the given
+ *                 linkName is unknown
+ * @throws {Error} If one or more of the params given isn't used in the URL
+ *                 template
+ *
+ * @ngInject
  */
-var ROUTES = {
-  'account.settings': 'account/settings',
-  'forgot-password': 'forgot-password',
-  'groups.leave': 'groups/:id/leave',
-  'groups.new': 'groups/new',
-  'help': 'docs/help',
-  'signup': 'signup',
-  'search.tag': 'search?q=tag:":tag"',
-  'user': 'u/:user',
-};
+function serviceUrl(annotationUI, store) {
 
-/**
- * A service which maps route names to URLs on the Hypothesis annotation
- * service.
- */
-// @ngInject
-function serviceUrl(settings) {
-  return function (route, params) {
-    params = params || {};
+  store.links()
+    .then(function updateLinks(links) {
+      annotationUI.updateLinks(links);
+    })
+    .catch(function() {
+      // We catch rejected promises here in order to silence
```

Comment:
One reason it is still worth logging the error here is that it might not be a failed HTTP request. eg. If something fails before or after the call is made.